src/download.rs

```diff
@@ -239,15 +247,32 @@ pub fn write_image(source: &mut ImageSource, dest: &mut File, decompress: bool)
         }
     };
 
-    // Cache the first MiB of input and write zeroes instead.  This ensures
-    // that the disk image can't be used accidentally before its GPG signature
-    // is verified.
+    // Read the first MiB of input and, if requested, check it against the
+    // image's formatted sector size.
     let mut first_mb: [u8; 1024 * 1024] = [0; 1024 * 1024];
-    dest.write_all(&first_mb)
-        .chain_err(|| "clearing first MiB of disk")?;
     decompress_reader
         .read_exact(&mut first_mb)
         .chain_err(|| "decoding first MiB of image")?;
+    // Were we asked to check sector size?
+    if let Some(expected) = expected_sector_size {
+        // Can we derive one from source data?
+        if let Some(actual) = detect_formatted_sector_size(&first_mb) {
```

Comment:
If we're asked to expect a sector size, shouldn't we fail if we can't query the sector size from the image?

Reply:
It's a best-effort check to improve UX.  We don't currently detect sector size on non-EFI platforms.